libbeat/autodiscover/autodiscover.go

```diff
@@ -66,7 +67,14 @@ type Autodiscover struct {
 }
 
 // NewAutodiscover instantiates and returns a new Autodiscover manager
-func NewAutodiscover(name string, pipeline beat.Pipeline, adapter Adapter, config *Config) (*Autodiscover, error) {
+func NewAutodiscover(
+	name string,
+	pipeline beat.Pipeline,
+	factory cfgfile.RunnerFactory,
+	selector EventSelector,
+	configurer EventConfigurer,
```

Comment:
selector/configurer are the same for all Beats. How about removing these altogether? I did only introduce them to keep the overall flexiblity that the Adapter did provide. @exekias Is there a reason/use-case to not remove those?

Reply:
I cannot think of any current use case so +1 to removing these. We can always add it when we need it

Reply:
How would they be removed -- does this mean that `queryConfigFrom` is the only implementor of `EventConfigurer`, and we can remove the interface and simply refer to the concrete struct? Or are you saying that the concrete type is also somehow redundant and we can remove this parameter entirely? Removing the interface would certainly resolve the associated documentation concerns, so I'm inclined to be in favor of it :-)

Does "selector" mean the same thing as "configurer" in this context or are you referring to two different values?

Reply:
Checking all Beats code the implementation is always the same. The `Configurer` always returns `[]string{ "config" }`, and the selector always reads the field `event["config"]`. In this sense the configurer and selector need to be configured together to be sound. I already combined the two into one interface in this PR. But given that all call sides always use `"config"`, I'm thinking to remove the interfaces completely.